models/enterprise_sonic/snmp/sonic_snmp.yml

```diff
@@ -2,52 +2,281 @@
-      choices:
-      - merged
-      - replaced
-      - overridden
-      - deleted
-      default: merged
+    module: sonic_snmp
+    version_added: 3.3.1
+    short_description: Manage SNMP configuration on SONiC
+    description:
+      - This module provides configuration management of SNMP for devices running SONiC
+    author: Aida Shumburo (@aida-shumburo)
+    options:
+      config:
+        description:
+          - SNMP configuration
+        type: dict
+        suboptions:
+          agentaddress:
+            description:
+              - List of SNMP agent address configuration
+              - I(interface) and I(vrf) are mutually exclusive.
+            type: list
+            elements: dict
+            suboptions:
+              ip:
+                description:
+                  - IPv4 or IPv6 address
+                type: str
+                required: true
+              port:
+                description:
+                  - UDP listening port, range 1024-65535
+                type: int
+              interface:
+                description:
+                  - Name of interface
+                type: str
+              vrf:
+                description:
+                  - Name of VRF
+                type: str
+          community:
+            description:
+              - List of SNMP community configuration
+            type: list
+            elements: dict
+            suboptions:
+              group:
+                description:
+                  - Name of community group
+                type: str
+              name:
+                description:
+                  - Name of community
+                type: str
+                required: true
+          engine:
+            description:
+              - SNMP engine ID
+              - Octet (hex) string with colon separated, 5 to 32 octets
+            type: str
+          user:
+            description:
+              - List of SNMP user configuration
+            type: list
+            elements: dict
+            suboption:
+              group:
+                description:
+                  - Name of user group
+                type: str
+              name:
+                description:
+                  - Name of user
+                type: str
+                required: true
+              auth:
+                description:
+                  - User authentication configuration
+                type: dict
+                suboptions:
+                  auth_type:
+                    description:
+                      - Authentication type
+                    type: str
+                    choices:
+                      - md5
+                      - sha
+                  key: 
+                    description: 
+                      - Authentication key
+                    type: str
+              priv:
+                description:
+                  - Privacy configuration
+                type: dict
+                suboptions:
```

Comment:
Sorry I missed this earlier: The situation for the "priv" sub-options is the same as for the "auth" sub-options. The two types of key (aes and des) are mutually exclusive. The same type of structure should be used here as for auth:

- provide a "priv_type" option with choices of "aes" or "des".
- provide one "key" option to hold the key for the specified priv_type.

Note: Both of these keys (auth and priv) must be specified in a playbook in either "encrypted" or "unencrypted" form based on the value of the "encrypted" option. It may be useful to specify that here. I am adding a suggestion below to clarify this point for users.